cpp/src/arrow/buffer.h

```diff
@@ -254,20 +252,51 @@ class ARROW_EXPORT Buffer {
   /// `data()` pointer.  Otherwise, you'll have to `View()` or `Copy()` it.
   bool is_cpu() const { return is_cpu_; }
 
+  /// \brief Whether the buffer is mutable
+  ///
+  /// If this function returns true, you are allowed to modify buffer contents
+  /// using the pointer returned by `mutable_data()` or `mutable_address()`.
+  bool is_mutable() const { return is_mutable_; }
+
   const std::shared_ptr<Device>& device() const { return memory_manager_->device(); }
 
   const std::shared_ptr<MemoryManager>& memory_manager() const { return memory_manager_; }
 
   std::shared_ptr<Buffer> parent() const { return parent_; }
 
-  // Convenience functions
+  /// \brief Get a RandomAccessFile for reading a buffer
+  ///
+  /// The returned file object reads from this buffer's underlying memory.
   static Result<std::shared_ptr<io::RandomAccessFile>> GetReader(std::shared_ptr<Buffer>);
+
+  /// \brief Get a OutputStream for writing to a buffer
+  ///
+  /// The buffer must be mutable.  The returned stream object writes into the buffer's
+  /// underlying memory (but it won't resize it).
```

Comment:
It might be worthwhile to duplicate this parenthesis in the doccomment for `MemoryManager::GetBufferWriter`

Reply:
Will do.